geode-core/src/main/java/org/apache/geode/internal/security/shiro/JMXShiroAuthenticator.java

```diff
@@ -37,6 +37,7 @@
 public class JMXShiroAuthenticator implements JMXAuthenticator, NotificationListener {
 
   private final SecurityService securityService;
+  private int numOfJMXConnections = 0;
```

Comment:
is this thread safe? Say two users logs in on different thread, I guess you would fail to log out the first user.